src/rgw/rgw_json_enc.cc

```diff
@@ -987,7 +987,9 @@ void RGWZone::decode_json(JSONObj *obj)
   JSONDecoder::decode_json("bucket_index_max_shards", bucket_index_max_shards, obj);
   JSONDecoder::decode_json("read_only", read_only, obj);
   JSONDecoder::decode_json("tier_type", tier_type, obj);
-  JSONDecoder::decode_json("sync_from_all", sync_from_all, obj);
+  if (false == JSONDecoder::decode_json("sync_from_all", sync_from_all, obj)){
```

Comment:
`JSONDecoder` has another overload that takes a `default_val` that looks perfect for this. see https://github.com/ceph/ceph/blob/master/src/common/ceph_json.h#L125

```c++
JSONDecoder::decode_json("sync_from_all", sync_from_all, true, obj);
```
(if that doesn't compile, we might need to change the `T& default_val` arg to `const T& default_val` in that overload)

Reply:
updated